src/components/modebar/buttons_config.js

```diff
@@ -0,0 +1,257 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+/**
+ * Modebar buttons configuration
+ *
+ * @param {string} category button category depending on e.g. plot type
+ * @param {string} group button group ('ext', 'drag', 'zoom', 'hover')
+ * @param {string} title text that appears while hovering over the button
+ * @param {string} icon name of the svg icon associated with the button
+ * @param {string} [gravity] icon positioning
+ * @param {string} click name of the modebar click handler associated with the button
+ * @param {string} [attr] attribute associated with button,
+ *                        use this with 'val' to keep track of the state
+ * @param {*} [val] initial 'attr' value,
+ *                  can be a function of graphInfo
+ * @param {boolean} [toggle] is the button a toggle button?
+ *
+ */
+
+module.exports = {
+
+    // for all plot types
+    toImage: {
+        category: 'all',
+        group: 'ext',
+        title: 'download plot as a png',
+        icon: 'camera',
+        click: 'toImage'
```

Comment:
What about dry-ness? The list of buttons groups was repeated in three places before. Now only once.